files/disable_inactive_freeipa_users.sh

```diff
@@ -69,59 +69,76 @@ else
-      sed --quiet 's/\s*Last successful authentication:\s*//p')
-    # In the event that all timestamps are invalid (N/A) we don't want
-    # to disable users since we don't know the last time they
-    # authenticated.
+    # Do an LDAP query to get the timestamps corresponding to the
+    # user's creation time and the user's last authentication time.
+    ldapsearch_output=$(ldapsearch -Y GSSAPI -b "$searchbase" "uid=$user" createTimestamp krbLastSuccessfulAuth)
+
+    # Extract the user's creation timestamp from the LDAP output.
     #
-    # TODO: After disable_time has elapsed with the "KDC:Disable Last
-    # Success" feature disabled we can start disabling users with all
-    # authentication timestamps invalid (N/A).  See #74 for more
-    # details.
-    all_timestamps_invalid=true
-    disable_user=true
-    for timestamp in $timestamps; do
-      # Do we have a valid timestamp in the format that the command
-      # ipa user-status uses (YYYYMMDDHHMMSSZ)?
-      if [[ $timestamp =~ ^([[:digit:]]{4})([[:digit:]]{2})([[:digit:]]{2})([[:digit:]]{2})([[:digit:]]{2})([[:digit:]]{2})Z$ ]]; then
-        # At least one timestamp is valid.
-        all_timestamps_invalid=false
-
-        # 1. Reformat the timestamp to ISO 8601 format so the date
-        # command can understand it.
-        # 2. Use the date command to convert the ISO 8601 timestamp to
-        # seconds since the epoch.
-        timestamp=${BASH_REMATCH[1]}-${BASH_REMATCH[2]}-${BASH_REMATCH[3]}T${BASH_REMATCH[4]}:${BASH_REMATCH[4]}:${BASH_REMATCH[6]}Z
-        timestamp=$(date --date="$timestamp" +%s)
-
-        # If the timestamp postdates the deadline then the user is
-        # active and should not be disabled.
-        if ((timestamp > disable_deadline)); then
-          disable_user=false
+    # --quiet means no printing unless the p command is used
+    create_timestamp=$(sed --quiet 's/^createTimestamp:\s//p' <<< "$ldapsearch_output")
+
+    # Do we have a valid timestamp in the format that IPA and LDAP use
+    # (YYYYMMDDHHMMSSZ)?
+    if [[ $create_timestamp =~ ^([[:digit:]]{4})([[:digit:]]{2})([[:digit:]]{2})([[:digit:]]{2})([[:digit:]]{2})([[:digit:]]{2})Z$ ]]; then
+      # 1. Reformat the timestamp to ISO 8601 format so the date
+      # command can understand it.
+      # 2. Use the date command to convert the ISO 8601 timestamp to
+      # seconds since the epoch.
+      create_timestamp=${BASH_REMATCH[1]}-${BASH_REMATCH[2]}-${BASH_REMATCH[3]}T${BASH_REMATCH[4]}:${BASH_REMATCH[4]}:${BASH_REMATCH[6]}Z
+      create_timestamp=$(date --date="$create_timestamp" +%s)
+
+      # This if skips any users that were created recently but have
+      # not yet logged in.  We don't want to disable their access yet.
+      if ((create_timestamp < disable_deadline)); then
+        # Extract the user's last authentication timestamp from the
+        # LDAP output.  Note that this timestamp may not exist if the
+        # user has never logged in.
```

Comment:
```suggestion
        # User has never logged in.
```

Reply:
Why should we randomly capitalize this one noun?  I am missing your point.

Reply:
...we shouldn't. I think I lost track of where I was and read this as a standalone sentence. Apologies!

Reply:
No worries.